tangelo/tangelo/__init__.py

```diff
@@ -267,3 +267,42 @@ def converter(*pargs, **kwargs):
 
         return converter
     return wrap
+
+
+localModuleCache = None
+
+
+def import_local(module, path='.'):
+    """
+    Import a module from a path relative to the module that called this
+    function.  If this module has changed since the last load, reimport it.
+    This will use the ModuleCase instance stored in localModuleCache.  If this
+    has not been set, a new module cache will be created.  The loaded module
+    is added to the caller's global namespace (just like import would do).
+    This functionally acts like 'import (module)' except that it will reload it
+    if the module's python file has changed.
+
+    :param module: the name of the module to import.  This needs to be a simple
+                   module name without any periods in it, and the file
+                   (module).py must exist in the relative path.
+    :param path: the relative path to the directory of the script with the
+                 calling function.  An absolute path can also be specified,
+                 but it isn't recommended.
```

Comment:
Is it ok with you guys if we discuss this at today's meeting?  If so I'll
make sure to review the PR by then.

On Tue, Sep 8, 2015 at 2:58 PM Jeffrey Baumes <notifications@github.com>
wrote:

> In tangelo/tangelo/__init__.py
> <https://github.com/Kitware/tangelo/pull/513#discussion_r38965756>:
>
> > +def import_local(module, path='.'):
> > +    """
> > +    Import a module from a path relative to the module that called this
> > +    function.  If this module has changed since the last load, reimport it.
> > +    This will use the ModuleCase instance stored in localModuleCache.  If this
> > +    has not been set, a new module cache will be created.  The loaded module
> > +    is added to the caller's global namespace (just like import would do).
> > +    This functionally acts like 'import (module)' except that it will reload it
> > +    if the module's python file has changed.
> > +
> > +    :param module: the name of the module to import.  This needs to be a simple
> > +                   module name without any periods in it, and the file
> > +                   (module).py must exist in the relative path.
> > +    :param path: the relative path to the directory of the script with the
> > +                 calling function.  An absolute path can also be specified,
> > +                 but it isn't recommended.
>
> Perhaps add something like "because absolute paths will not be portable."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Kitware/tangelo/pull/513/files#r38965756>.
>